### PR TITLE
Fix logger

### PR DESCRIPTION
**Pull request by yiliu30**
## Type of Change

Enhancement.

## Description

Enhance log funcs to show the filename and line number.

## How has this PR been tested?

Pre-CI

## Dependency Change?

None

### CODE REVIEW

File: test/3x/common/test_logger.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    {"msg": {("bert", "embedding"): {"weight": {"dtype": ["unint8", "int8"]}}}},
+    {"msg": {("bert", "embedding"): {"op": ("a", "b")}}},
+    # the following log will not be prettified
+    [{"msg": "call logger warning function"}, {"msg2": "done"}],
+    ({"msg": "call logger warning function"}, {"msg2": "done"}),
+    ({"msg": [{"sub_msg": "call logger"}, {"sub_msg2": "call warning function"}]}, {"msg2": "done"}),
+]
+
+
+class TestLogger(unittest.TestCase):
```
Comment (Kaihui-intel) [inline test/3x/common/test_logger.py:30]: Improvements can be made to ensure that the exception handling works, such as passing empty messages and exceptional messages at different levels.

**Review by Kaihui-intel — APPROVED** [review]

File: test/3x/torch/test_logger.py [hunk 2]
```diff
@@ -3,6 +3,21 @@
 
 from neural_compressor.common.utils import logger
 
+log_msg_lst = [
+    "call logger log function.",
```
Comment (yuwenzho) [inline test/3x/torch/test_logger.py:7]: `3x/torch/test_logger.py`, do we need additional test for each framework?

Comment (chensuyue) [reply test/3x/torch/test_logger.py:7]: I have the same question.

Comment (yiliu30) [reply test/3x/torch/test_logger.py:7]: No, the logger tests are framework-agnostic. I will remove `3x/torch/test_logger.py` later.
More details about the impact of `3x/common` can be found here https://github.com/intel/neural-compressor/discussions/1445#discussioncomment-8227945.

Comment (yiliu30) [reply test/3x/torch/test_logger.py:15]: I removed it in [f290f07](https://github.com/intel/neural-compressor/pull/1583/commits/f290f07f009d0de245cdfa3726e0546ddf0b888b).

File: test/3x/common/test_logger.py [hunk 3]
```diff
@@ -29,6 +51,18 @@ def test_logger(self):
         logger.warning(({"msg": "call logger warning function"}, {"msg2": "done"}))
         logger.warning(({"msg": [{"sub_msg": "call logger"}, {"sub_msg2": "call warning function"}]}, {"msg2": "done"}))
 
+    def test_logger_func_and_pretty_dict(self):
+        from neural_compressor.common.utils import debug, error, fatal, info, log, warning
```
Comment (yuwenzho) [inline test/3x/common/test_logger.py:55]: Which format is more recommended?  `from neural_compressor.common.utils import debug` or `from neural_compressor.common import logger`?

Comment (yiliu30) [reply test/3x/common/test_logger.py:55]: The purpose of `from neural_compressor.common import debug` is to make the logger dict pretty.
Here is a demo:
```python
FP32 = {"weight": {"dtype": ["fp32"]}, "activation": {"dtype": ["fp32"]}}

from neural_compressor.common import logger
logger.info(FP32)
# 2024-01-31 09:21:19 [INFO][log_demo.py:4] {'weight': {'dtype': ['fp32']}, 'activation': {'dtype': ['fp32']}}


logger.info("---------------------")
from neural_compressor.common import info
info(FP32)
# 2024-01-31 09:21:19 [INFO][log_demo.py:8] {
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]     'weight': {
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]         'dtype': [
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]             'fp32'
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]         ]
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]     },
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]     'activation': {
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]         'dtype': [
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]             'fp32'
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]         ]
# 2024-01-31 09:21:19 [INFO][log_demo.py:8]     }
# 2024-01-31 09:21:19 [INFO][log_demo.py:8] }
```

Comment (yiliu30) [reply test/3x/common/test_logger.py:55]: Perhaps we should move the functionality of the pretty dict to logger and only maintain one usage: `from neural_compressor.common import logger`.

**Review by yuwenzho — APPROVED** [review]

File: test/3x/common/test_logger.py (hunk 1 above)
Comment (yiliu30) [reply test/3x/common/test_logger.py:30]: The Logger invokes the built-in `logging` package directly, which  should have a comprehensive test.

File: test/3x/common/test_logger.py (hunk 3 above)
Comment (yiliu30) [reply test/3x/common/test_logger.py]: Only keep `from neural_compressor.common import logger`.

**Comment by yiliu30** [comment]
The pre-commit update in the documents.

**Pull request merged**